trio/_highlevel_socket.py

```diff
@@ -1,13 +1,18 @@
 # "High-level" networking interface
+from __future__ import annotations
 
 import errno
 from contextlib import contextmanager
+from typing import TYPE_CHECKING
 
 import trio
 from . import socket as tsocket
 from ._util import ConflictDetector, Final
 from .abc import HalfCloseableStream, Listener
 
+if TYPE_CHECKING:
+    from ._socket import _SocketType as SocketType
```

Comment:
I think we should come up with a policy around runtime introspection. This can stay for now while we figure it out, but I don't know if the current direction this is all heading (you cannot inspect trio's types at runtime for some functions) is great. I don't know if there's any solid reasons to introspect at runtime either, so that's a thing.

Reply:
yeah, and add isort+flake8-annotations to CI cause imports are a mess.

Reply:
opened https://github.com/python-trio/trio/issues/2687